cfme/networks/cloud_tenant.py

```diff
@@ -0,0 +1,72 @@
+import attr
+
+from cfme.cloud.tenant import TenantDetailsView
+from cfme.common import Taggable
+from cfme.modeling.base import BaseCollection, BaseEntity, parent_of_type
+from cfme.networks import ValidateStatsMixin
+from cfme.utils import version
+from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+
+@attr.s
+class CloudTenant(Taggable, BaseEntity, ValidateStatsMixin):
+    """Class representing cloud tenants"""
+    in_version = ('5.10', version.LATEST)
+    category = "networks"
+    string_name = 'CloudTenant'
+    quad_name = None
+    db_types = ["CloudTenant"]
+
+    name = attr.ib()
+    provider_obj = attr.ib(default=None)
+
+    @property
+    def provider(self):
+        from cfme.networks.provider import NetworkProvider
+        return parent_of_type(self, NetworkProvider)
+
+    @property
+    def num_cloud_subnets(self):
```

Comment:
There are many more of these fields on the page, I don't think its a good pattern to try to implement these boilerplate methods to get the relationship values from the tables.

Reply:
I think it's a common pattern to do it like this @mshriver (we've seen it with other providers), but what you say makes sense. @aljazkosir maybe we can implement following method instead:

```python
def ui_details_value(property_name)
  return view.entities.relationships.get_text_of(property_name)
```

and then update ValidateStatsMixin to just make use of it. Should be quite trivial and indeed, as @mshriver suggests, quite less boilerplate.